## THE NULL MODEL CV SCORE USING ACCURACY AS METRIC WAS 0.57034
### We have to beat that

In [1]:
import pandas as pd
import numpy as np

In [2]:
train = pd.read_csv("../data/train.csv")
test = pd.read_csv("../data/test.csv")

In [3]:
feature_cols = ["keyword", "location"]

In [4]:
X = train[feature_cols]
y = train["target"]

## ACTION PLAN FOR THIS NOTEBOOK
- Impute missing values in the keyword and location columns
- Use this columns to create a predictive model

In [13]:
from sklearn.pipeline import make_pipeline
from catboost import CatBoostClassifier
from xgboost import XGBClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.impute import SimpleImputer
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import OrdinalEncoder
from sklearn.preprocessing import OneHotEncoder

In [6]:
X_cleaned = X.copy()
X_cleaned["location"] = X["location"].fillna("None")
X_cleaned["keyword"] = X["keyword"].fillna("None")

In [7]:
Encod = LabelEncoder()
X_Encod = X_cleaned.copy()
X_Encod["keyword"] = Encod.fit_transform(X_cleaned["keyword"])
X_Encod["location"] = Encod.fit_transform(X_cleaned["location"])

In [22]:
Catty = CatBoostClassifier(silent = True)
Loggy = LogisticRegression(penalty = "l1", max_iter = 1000)
XGB = XGBClassifier()

In [23]:
score = cross_val_score(Loggy, X_Encod, y, cv = 5, scoring = "accuracy").mean()

C:\Users\Adeoluwa Adeboye\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\Adeoluwa Adeboye\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\Adeoluwa Adeboye\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\Adeoluwa Adeboye\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)
C:\Users\Adeoluwa Adeboye\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be ch

In [24]:
print(score)

0.5462929035058536


## ITS SEEMS THIS MODEL #0.54629 IS NOT EVEN AS GOOD HAS THE NULL MODEL #0.5703